sql-db/hibernate-fulltext-search/src/main/java/io/quarkus/ts/hibernate/search/FruitResource.java

```diff
@@ -115,4 +120,51 @@ public Response search(@QueryParam("terms") String terms) {
                 .fetchAllHits();
         return Response.status(Response.Status.OK).entity(list).build();
     }
+
+    @GET
+    @Path("/search-using-metamodel-class")
+    @Transactional
+    @Blocking
+    public Response searchUsingMetamodelClasses(@QueryParam("terms") String terms) {
+        List<Fruit> list = searchSession.search(Fruit__.INDEX.scope(searchSession))
```

Comment:
What is "Fruit___"? Some class generated by Hibernate?

Reply:
Hibernate Search generates these classes so that you have type-safe behavior. Depending on the implemented interfaces, you are not allowed to invoke methods that doesn't make sense for respective field. e.g. you won't have averages for string values. Also, when you rename property, hardcoded string won't detect this change, while this generated class will detect and result in the compilation failure.